CMakeSettings.json

```diff
@@ -1,22 +1,34 @@
 ﻿{
-  "configurations": [
-    {
-      "name": "x64-Debug",
-      "generator": "Ninja",
-      "configurationType": "Debug",
-      "inheritEnvironments": [ "msvc_x64_x64" ],
-      "buildRoot": "${projectDir}\\out\\build\\${name}",
-      "installRoot": "${projectDir}\\out\\install\\${name}",
-      "cmakeCommandArgs": "",
-      "buildCommandArgs": "-v",
-      "ctestCommandArgs": "",
-      "variables": [
+    "configurations": [
         {
-          "name": "CMAKE_BUILD_TYPE",
-          "value": "Debug",
-          "type": "STRING"
+            "name": "x64-Debug",
+            "generator": "Ninja",
+            "configurationType": "Debug",
```

Comment:
I have seen the concept of "Debug" in a bunch of places now. Do we need to be building "Release" as well? If so, what types of optimizations are added for that configuration?

Reply:
when we run cmake with gcc on Linux, release is used by default.  This means we can't debug (no symbols are generated for debug).  It also means compiler will try to make as many optimizations as possible (that will depend on compiler)

On the other hand, on Visual Studio, Debug is the default build configuration, maybe because you can set breakpoints and use the run this file option to start debugging at any point.

The only reason we (or anyone) would like to build Debug output is for either debugging or to get code coverage (we need to mapping from symbols to src code)